### PR TITLE
[FLINK-34422][test] BatchTestBase uses MiniClusterExtension

### PR DESCRIPTION
**Pull request by zentol**
The setup of the table environment in this class ran before the MiniClusterExtension was set up and beforeEach was called, causing jobs to spawn separate mini clusters. Certain tests cases try running ~200 test cases in parallel, which overloaded the system.

### CODE REVIEW

**Comment by zentol** [comment]
@flinkbot run azure

File: flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/ProtobufSQLITCaseTest.java
Comment (XComp) [inline flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/ProtobufSQLITCaseTest.java:1]: Can we also fix the naming in a hotfix commit? `ProtobufSQLITCaseTest` -> `ProtobufSQLITCase`

File: flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/VeryBigPbProtoToRowTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 import org.apache.flink.formats.protobuf.testproto.VeryBigPbClass;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
```
Comment (XComp) [inline flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/VeryBigPbProtoToRowTest.java:23]: That one looks like it's accidentally added?

**Review by XComp — APPROVED** [review]
LGTM :+1:

File: flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/VeryBigPbProtoToRowTest.java (hunk 1 above)
Comment (zentol) [reply flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/VeryBigPbProtoToRowTest.java:23]: uuuh, jup.

File: flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/ProtobufSQLITCaseTest.java
Comment (zentol) [reply flink-formats/flink-protobuf/src/test/java/org/apache/flink/formats/protobuf/ProtobufSQLITCaseTest.java:1]: sure

**Pull request merged**